source/common/listener_manager/listener_impl.cc

```diff
@@ -323,16 +342,16 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
       open_connections_(std::make_shared<BasicResourceLimitImpl>(
           std::numeric_limits<uint64_t>::max(),
           listener_factory_context_->serverFactoryContext().runtime(), cx_limit_runtime_key_)),
-      local_init_watcher_(fmt::format("Listener-local-init-watcher {}", name),
-                          [this] {
-                            if (workers_started_) {
-                              parent_.onListenerWarmed(*this);
-                            } else {
-                              // Notify Server that this listener is
-                              // ready.
-                              listener_init_target_.ready();
-                            }
-                          }),
+      local_init_watcher_(std::make_shared<Init::WatcherImpl>(
+          fmt::format("Listener-local-init-watcher {}", name),
+          [this, target = listener_init_target_, workers_started = workers_started_]() {
```

Comment:
`this` might be dead if the shared_ptr watcher impl outlives it? you should probably explain the lifetime correctness.